components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java

```diff
@@ -263,28 +266,60 @@ public int getMaxLoginAttemptCount() {
 
     private String buildUrl(String defaultContext, Supplier<String> getValueFromFileBasedConfig) {
 
-        String url;
-        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
-            url = buildTenantQualifiedUrl(defaultContext);
+        if (log.isDebugEnabled()) {
+            log.debug("Building url for context: " + defaultContext);
+        }
+
+        String context = defaultContext;
+
+        String urlFromFileBasedConfig = getValueFromFileBasedConfig.get();
+        if (StringUtils.isNotBlank(urlFromFileBasedConfig)) {
+            // If the file based config is set, then we have to build the URL based on that.
+            if (log.isDebugEnabled()) {
+                log.debug("URL from file based config: " + urlFromFileBasedConfig);
+            }
+            context = urlFromFileBasedConfig;
+        }
+
+        if (isUrlAbsolute(context)) {
+            if (log.isDebugEnabled()) {
+                log.debug("URL: " + context + " is absolute. Returning it as the resolved URL.");
+            }
+            return context;
```

Comment:
There's a small concern here.
We should read from file only when it's not tenant url supported mode right.
If it's tenant url supported, context is always a path, that will be resolved to an absolute URL from the URL builder.

If tenant url not supported only we will have to worry about, the context read from the file is absolute or not.

Reply:
Even in the tenant qualified mode, we might have a situation where a user has defined a custom path. For example: instead of `authenthenticationendpoint/login.do` someone could define `authenthenticationendpoint/custom_login.do`

So when the facade returns in the tenant qualified mode, we can simply return 
`\t\<tenant_name>\authenthenticationendpoint/custom_login.do`

Reply:
I was actually thinking that should be resolved local to the tenant.
But, unlike other endpoints which are defined for the server, there's a case with related to authentication endpoint web app. That is, it allows to be externalized, which can mean server wide there can be an endpoint URL configured, where your above comment applies. Mode of the offering needs to be finalized then